iohub/multipagetiff.py

```diff
@@ -209,15 +209,17 @@ def _set_mm_meta(self):
                 for ch in self.mm_meta["Summary"]["ChNames"]:
                     self.channel_names.append(ch)
 
-            # dimensions based on mm metadata
-            # do not reflect final written dimensions
-            # these will change after data is loaded
             self.z_step_size = self.mm_meta["Summary"]["z-step_um"]
             self.height = self.mm_meta["Summary"]["Height"]
             self.width = self.mm_meta["Summary"]["Width"]
-            self.frames = self.mm_meta["Summary"]["Frames"]
-            self.slices = self.mm_meta["Summary"]["Slices"]
-            self.channels = self.mm_meta["Summary"]["Channels"]
+
+            # dimensions based on mm metadata
+            # do not reflect final written dimensions
+            # these set in _gather_index_maps
+            #
+            # self.frames = self.mm_meta["Summary"]["Frames"]
+            # self.slices = self.mm_meta["Summary"]["Slices"]
+            # self.channels = self.mm_meta["Summary"]["Channels"]
```

Comment:
Thanks for catching this @ziw-liu. I commented out these lines so they don't overwrite the dimensions set by `_gather_index_maps`